### PR TITLE
Jobs batch endpoints:

### PR DESCRIPTION
**Pull request by iskrypka-smartling**
1) addFileToJobBatchAsync
2) BatchItemStatusResponse

### CODE REVIEW

File: spec/openapi.yaml [hunk 1]
```diff
@@ -8358,7 +8427,212 @@ paths:
...
+    post:
+      summary: Add file to a batch asynchrounosly
+      description: >
+        This endpoint is actually a proxy for Upload File in the Files API, and
+        it works with these two differences:
+         1. It additionally gets `batchUid` as a path parameter.
+         2. It doesn't relay `localeIdsToAuthorize` to Files Api, therefore the file stays unauthorized. Instead, `localeIdsToAuthorize` are saved just for this batch session and are used when a file is being attached to a job.
+
+        When the first file is added to the new batch, it becomes an
+        `ADDING_FILES` status. Adds original source content to a Job batch in
```
Comment (hgronberg-smartling) [inline spec/openapi.yaml]: Change "it becomes an 'adding_files' status" to "the batch changes to 'ADDING_FILES' status"

File: spec/openapi.yaml [hunk 2]
```diff
@@ -8358,7 +8427,212 @@ paths:
...
+
+        The curl example provided will upload your Java properties file directly
+        to a Smartling project, identified by the `projectId`, and attaches it
+        to a Job linked to `batchUid`. The response is returned right after the
+        content of a file is accepted by the Files API. After that, the Batch
+        service controls whether the file upload finished. Content is parsed,
+        all strings are ingested, and the file is then attached to the Job which
+        is linked to the batch for locales, described in `localeIdsToAuthorize`.
+
+        Endpoint returns response as soon as file uploaded to jobs batch server. File is uploaded to FILE API in separate thread.
```
Comment (hgronberg-smartling) [inline spec/openapi.yaml]: Change "as soon as file uploaded to" to "as soon as a file is uploaded to"

File: spec/openapi.yaml [hunk 3]
```diff
@@ -8358,7 +8427,212 @@ paths:
...
+                    it is uploaded. Authorization is happened only when job is
+                    authorized.
+                  items:
+                    type: string
+                  type: array
+                smartling.namespace:
+                  description: >
+                    This is only for accounts created after July 31st, 2015.
+                    Define a custom
+                    [namespace](https://developer.smartling.com/v2.0/docs/string-sharing-namespaces)
```
Comment (hgronberg-smartling) [inline spec/openapi.yaml]: The link should be https://help.smartling.com/hc/en-us/articles/360008143833

File: spec/openapi.yaml [hunk 4]
```diff
@@ -8358,7 +8427,212 @@ paths:
...
+                    cannot be changed. If you reupload the file, it will use the
+                    original charset, even if you change the directive. An error
+                    will be returned if this directive is used when uploading
+                    binary-format file types, such as Office or IDML files.
+                  type: string
+                'smartling.[command]':
+                  description: >
+                    Provides custom parser configuration for supported file
+                    types. See [Supported File
+                    Types](https://developer.smartling.com/docs/supported-file-types)
```
Comment (hgronberg-smartling) [inline spec/openapi.yaml]: link should be https://help.smartling.com/hc/en-us/articles/360007998893

File: spec/openapi.yaml [hunk 5]
```diff
@@ -9489,6 +9763,60 @@ components:
...
+                    fileUri:
+                      description: Uri of the file linked to the Batch
+                      type: string
+                    batchUid:
+                      description: The UID of the batch
+                      type: string
+                    batchItemId:
+                      description: The id of the batch item
+                      type: integer
+    BatchItemStatusResponse:
```
Comment (DmitryBochorishvili) [inline spec/openapi.yaml]: I would call it `JobBatchItemStatusResponse` to make it clearer what kind of batch item it's about

File: spec/openapi.yaml [hunk 6]
```diff
@@ -8193,6 +8193,75 @@ paths:
...
             -H "Authorization: Bearer $smartlingToken"
 
+  '/jobs-batch-api/v1/projects/{projectId}/batches/{batchUid}/items/{batchItemId}':
+    get:
+      summary: Get batch item status
+      description: >-
+        Returns batch item status.
+      tags:
+      - Job batches
+      operationId: getJobBatchItemStatus
```
Comment (DmitryBochorishvili) [inline spec/openapi.yaml]: It's great you're adding operationId!

**Pull request merged**